### ISSUE TITLE
Monkey King Tree Dance not behaving correctly when cast on Nature Prophet Sprout

### ISSUE THREAD

**Issue by SoumyajitKarmakar** [jeffhill/Dota2#1022]
### Ability name

Tree Dance, Sprout

### Description

I would like to report a MK tree dance bug which is observed when used on NP sprout trees. If MK is on the sprout trees and the duration expires then MK does not get stunned. If MK is on the sprout tree and NP uses Nature's Call to destroy all trees, even then MK is not stunned. Also there is an inconsistency, if quelling blade is used to cut down the sprout tree when MK is on it, then sometimes he gets stunned and sometimes he does not (check the video at 1:21).

PS. I have recorded the video in demo mode, hence instead of Match ID I have given the drive link to the video.

[link to drive](https://drive.google.com/file/d/1rH-sIjnK0z-pCJbIoQlhWQgwXYSmF3oN/view?usp=sharing)

### Example Match ID (and possibly Timestamp)

https://drive.google.com/file/d/1rH-sIjnK0z-pCJbIoQlhWQgwXYSmF3oN/view?usp=sharing

**Comment by real-vaLor** [comment]
> If MK is on the sprout trees and the duration expires then MK does not get stunned.
I dont think this is a "bug", the trees were never destroyed, they expired, so i dont think it makes sense for it to stun MK in that situation

For the rest I agree
